### ISSUE TITLE
FEI: no match for 'operator<< (GCC 5.2.1)

### ISSUE THREAD

**Issue by nschloe** [trilinos/Trilinos#20]
When configuring Trilinos with
```sh
cmake \
  .. \
  -DCMAKE_INSTALL_PREFIX=/usr \
  -DCMAKE_VERBOSE_MAKEFILE=ON \
  -DCMAKE_BUILD_TYPE=None \
  -DCMAKE_C_COMPILER=mpicc \
  -DCMAKE_CXX_COMPILER=mpicxx \
  -DCMAKE_Fortran_COMPILER=mpif90 \
  -DBUILD_SHARED_LIBS:BOOL=ON \
  -DCMAKE_SKIP_RPATH:BOOL=ON \
  -DTrilinos_LIBRARY_NAME_PREFIX:STRING=trilinos_ \
  -DTrilinos_INSTALL_INCLUDE_DIR:PATH=include/trilinos/ \
  -DTrilinos_USE_GNUINSTALLDIRS:BOOL=ON \
  -DTrilinos_ENABLE_DEVELOPMENT_MODE:BOOL=OFF \
  -DTrilinos_ENABLE_ALL_PACKAGES:BOOL=ON \
  -DTrilinos_ENABLE_SECONDARY_TESTED_CODE:BOOL=ON \
  -DTrilinos_ASSERT_MISSING_PACKAGES:BOOL=OFF \
  -DTrilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON \
  -DTrilinos_ENABLE_Didasko:BOOL=OFF \
  -DTrilinos_ENABLE_Gtest:BOOL=OFF \
  -DTrilinos_ENABLE_CTrilinos:BOOL=OFF \
  -DTrilinos_ENABLE_ForTrilinos:BOOL=OFF \
  -DTrilinos_ENABLE_Mesquite:BOOL=OFF \
  -DTrilinos_ENABLE_MOOCHO:BOOL=OFF \
  -DTrilinos_ENABLE_Optika:BOOL=OFF \
  -DTrilinos_ENABLE_Phdmesh:BOOL=OFF \
  -DTrilinos_ENABLE_PyTrilinos:BOOL=OFF \
  -DTrilinos_ENABLE_Sundance:BOOL=OFF \
  -DTrilinos_ENABLE_STKClassic:BOOL=OFF \
  -DTrilinos_ENABLE_STKDoc_tests:BOOL=OFF \
  -DTrilinos_ENABLE_STKSearch:BOOL=OFF \
  -DTrilinos_ENABLE_STKUnit_tests:BOOL=OFF \
  -DTrilinos_ENABLE_ThreadPool:BOOL=OFF \
  -DTrilinos_ENABLE_EXAMPLES:BOOL=OFF \
  -DTrilinos_ENABLE_TESTS:BOOL=OFF \
  -DSEACAS_ENABLE_NETCDF4_SUPPORT:BOOL=ON \
  -DSEACASExodus_ENABLE_MPI:BOOL=OFF \
  -DTPL_ENABLE_BinUtils:BOOL=ON \
  -DTPL_ENABLE_Boost:BOOL=ON \
  -DTPL_ENABLE_HDF5:BOOL=OFF \
  -DTPL_ENABLE_Matio:BOOL=OFF \
  -DTPL_ENABLE_MATLAB:BOOL=OFF \
  -DTPL_ENABLE_MPI:BOOL=ON \
  -DTPL_ENABLE_MUMPS:BOOL=ON \
  -DTPL_ENABLE_ParMETIS:BOOL=OFF \
  -DTPL_ENABLE_Scotch:BOOL=ON \
  -DTPL_Scotch_INCLUDE_DIRS:PATH=/usr/include/scotch/ \
  -DTPL_ENABLE_SuperLU:BOOL=ON \
  -DSuperLU_INCLUDE_DIRS:PATH=/usr/include/superlu/ \
  -DTPL_ENABLE_TBB:BOOL=ON \
  -DTPL_ENABLE_X11:BOOL=OFF \
  -DTPL_ENABLE_Zlib:BOOL=ON
```
(can perhaps be stripped down) and when using GCC 5.2.1, one gets the compile error
```
[...]
/«PKGBUILDDIR»/packages/fei/test_utils/snl_fei_tester.cpp:526:65: error: no match for 'operator<<' (operand types are 'std::basic_ostream<char>' and 'std::ostringstream {aka std::__cxx11::basic_ostringstream<char>}')
     fei::console_out() << "ERROR opening solution output file " << fileName << FEI_ENDL;
                                                                 ^
/«PKGBUILDDIR»/packages/fei/test_utils/snl_fei_tester.cpp:526:65: note: candidate: operator<<(int, int) <built-in>
/«PKGBUILDDIR»/packages/fei/test_utils/snl_fei_tester.c
```
The full log is [here](https://launchpadlibrarian.net/214282732/buildlog_ubuntu-wily-amd64.trilinos_12.3~201508121416-wily1_BUILDING.txt.gz).

**Comment by aprokop** [comment]
Nico, which version of Trilinos are you using? This was fixed in 75015192384a

**Comment by nschloe** [comment]
This was with 12.2.1. Tested again with current `master` and it all works. Thanks for checking back!
